internal/cmd/grants.go

```diff
@@ -286,35 +293,78 @@ func removeGrant(cli *CLI, cmdOptions grantsCmdOptions) error {
 		return err
 	}
 
-	if grants.Count == 0 && !cmdOptions.Force {
+	switch {
+	case cmdOptions.Force:
+		// noop
+	case grants.Count == 0:
 		return Error{Message: "Grant not found"}
+	case grants.Count > 1:
+		var sb strings.Builder
+		sb.WriteString("Multiple grants found for")
+
+		switch {
+		case user != 0:
+			fmt.Fprintf(&sb, " %q", cmdOptions.UserName)
+		case group != 0:
+			fmt.Fprintf(&sb, " %q", cmdOptions.GroupName)
+		}
+
+		fmt.Fprintf(&sb, " %q", cmdOptions.Destination)
+		if cmdOptions.Resource != "" {
+			fmt.Fprintf(&sb, " resource %q", cmdOptions.Resource)
+		}
+
+		if cmdOptions.Role != "" {
+			fmt.Fprintf(&sb, " role %q", cmdOptions.Role)
+		}
+
+		sb.WriteString(". Continue?")
+
+		var confirmRemove bool
+		if err := survey.AskOne(&survey.Confirm{Message: sb.String()}, &confirmRemove, cli.surveyIO); err != nil {
+			return err
+		}
+
+		if !confirmRemove {
+			return Error{Message: "operation aborted"}
+		}
 	}
 
 	for _, g := range grants.Items {
 		logging.Debugf("call server: delete grant %s", g.ID)
-		err := client.DeleteGrant(ctx, g.ID)
-		if err != nil {
-			if api.ErrorStatusCode(err) == 403 {
-				logging.Debugf("%s", err.Error())
-				return Error{
-					Message: "Cannot revoke grants: missing privileges for DeleteGrant",
+		if !cmdOptions.DryRun {
```

Comment:
`--dryrun` allows the user to preview changes before actually running it against the server. It's particularly useful with removals since it's a destructive process but can be useful for `add` as well